flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/slot/TaskSlotUtils.java

```diff
@@ -69,6 +70,22 @@ public static ResourceProfile createDefaultSlotResourceProfile() {
 		return DEFAULT_RESOURCE_PROFILE;
 	}
 
+	public static ResourceProfile createTotalResourceProfile(int numberOfSlots) {
+		return createTotalResourceProfile(numberOfSlots, DEFAULT_RESOURCE_PROFILE);
+	}
+
+	public static ResourceProfile createTotalResourceProfile(int numberOfSlots, ResourceProfile defaultResourceProfile) {
+		ResourceProfile result = defaultResourceProfile;
+		while (numberOfSlots > 1) {
+			result = result.merge(result);
+			if (numberOfSlots % 2 != 0) {
+				result = result.merge(defaultResourceProfile);
+			}
+			numberOfSlots = numberOfSlots / 2;
```

Comment:
I would keep it as simple loop merging `defaultResourceProfile` `numberOfSlots - 1` times for simpler readability.